File: Bymed.Application/PageContent/PageContentDto.cs
namespace Bymed.Application.PageContent;

public sealed record PageMetadataDto
{
    public string? MetaTitle { get; init; }
    public string? MetaDescription { get; init; }
    public string? OgImage { get; init; }
}

public sealed record PageContentDto
{
    public required Guid Id { get; init; }
    public required string Slug { get; init; }
    public required string Title { get; init; }
    public required string Content { get; init; }
    public PageMetadataDto Metadata { get; init; } = new();
    public DateTime? PublishedAt { get; init; }
    public bool IsPublished { get; init; }
    public DateTime CreationTime { get; init; }
}

File: bymed-admin/src/app/shared/models/page-content.dto.ts
/** Mirrors API `PageMetadataDto` / `PageContentDto` (camelCase JSON). */
export interface PageMetadataDto {
  readonly metaTitle?: string | null;
  readonly metaDescription?: string | null;
  readonly ogImage?: string | null;
}

export interface PageContentSummaryDto {
  readonly id: string;
  readonly slug: string;
  readonly title: string;
  readonly content: string;
  readonly metadata: PageMetadataDto;
  readonly publishedAt?: string | null;
  readonly isPublished: boolean;
  readonly creationTime: string;
}
